app/main.py

```diff
@@ -2,0 +1,26 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health:"
+                f" {self.health}, Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(victim: Herbivore) -> None:
+        if isinstance(victim, Herbivore) and not victim.hidden:
```

Comment:
The `isinstance` check for `Herbivore` is redundant because the `bite` method already specifies that the `victim` parameter is of type `Herbivore`. You can safely remove this check.